### PR TITLE
Hilsim xplane build

### PR DESCRIPTION
**Pull request by Inspirati**

### CODE REVIEW

**Comment by wpremerlani** [comment]
Robert,
Thank you. I appreciate your continued interest and support.
Best regards,
Bill

**Comment by phollands** [comment]
Mark, thanks for the PR. I'll  test it out and review starting on Monday (GMT).. Best wishes, Pete

File: Tools/MatrixPilot-SIL/UDBSocket.h [hunk 1]
```diff
@@ -9,10 +9,11 @@
 #ifndef MatrixPilot_SIL_SIL_sockets_h
 #define MatrixPilot_SIL_SIL_sockets_h
 
-#if (!defined(WIN32) || defined(WIN))
+#if (!defined(WIN32) || defined(WIN) || !defined(_MSC_VER))
 #include <stdint.h>
 #else
-#include "stdint-win.h"
+//#include "stdint-win.h" // only for MS Visual C++ builds
```
Comment (phollands) [inline Tools/MatrixPilot-SIL/UDBSocket.h]: Commented can go.

File: libDCM/hilsim.c [hunk 2]
```diff
@@ -21,10 +21,121 @@
```
Comment (phollands) [inline libDCM/hilsim.c]: I would see this file as not being part of libDCM which in my mind is the IMU portion of MatrixPilot. How about putting it back libUDB if it has to go somewhere more mainstream than in Tools/....

File: libDCM/hilsim.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// This file is part of MatrixPilot.
+//
+//    http://code.google.com/p/gentlenav/
+//
+// Copyright 2009-2011 MatrixPilot Team
+// See the AUTHORS.TXT file for a list of authors of MatrixPilot.
```
Comment (phollands) [inline libDCM/hilsim.h]: Bump copyright to 2016.

**Comment by phollands** [comment]
Just to be clear on my summary comment above; I was not able to get the code to compile using standard gcc and Make. And with regard to X-Code projects, it did not compile for me as well (but I am not very familiar with X-Code).

**Comment by phollands** [comment]
Robert, FYI: I'm back on my normal MAC after travelling.  I have started testing this PR in SILSIM on the Mac and X-Plane 10.

File: Config/Cessna/options_magnetometer.h [hunk 4]
```diff
@@ -21,7 +21,7 @@
 // To do: select magnetometer type, set MAGNETICDECLINATION,
 // and select orientation of the magnetometer, and remove the next 3 lines.
 #if (MAG_YAW_DRIFT == 1 && HILSIM != 1)
-#warning "Check magnetometer options."
+//#warning "Check magnetometer options."
```
Comment (Inspirati) [inline Config/Cessna/options_magnetometer.h]: I agree, and have never much liked the plethora of warnings this can produce on the default options build. If we choose to keep this warning, I'd be in favour of at least moving it into the magnetometer source file and then at least it'll only chime in once. That said, for the moment I'd like to leave such a change to another PR.

File: MatrixPilot/MAVLink.c [hunk 5]
```diff
@@ -650,7 +650,7 @@ static void handleMessage(void)
 	handling_of_message_completed |= MAVParamsHandleMessage(handle_msg);
 	handling_of_message_completed |= MAVMissionHandleMessage(handle_msg);
 	handling_of_message_completed |= MAVFlexiFunctionsHandleMessage(handle_msg);
-	handling_of_message_completed |= MAVFTPHandleMessage(handle_msg);
+//	handling_of_message_completed |= MAVFTPHandleMessage(handle_msg); // WIP - RobD
```
Comment (Inspirati) [inline MatrixPilot/MAVLink.c]: Agreed. And fully agree with the philosophy TomP put forward. In general with code that needs to be deemed airworthy, 'dead code' is quite a no no. However in this particular instance, I'd like to keep this here as more a comment alluding to the fact that such functionality is on the table as work in progress, and as I believe it adds a little value as a place holder for where such would be called in.

File: MatrixPilot/MAVLink.c [hunk 6]
```diff
@@ -1074,7 +1074,7 @@ void mavlink_output_40hz(void)
 	MAVParamsOutput_40hz();
 	MAVMissionOutput_40hz();
 	MAVFlexiFunctionsOutput_40hz();
-	MAVFTPOutput_40hz();
+//	MAVFTPOutput_40hz(); // WIP - RobD
```
Comment (Inspirati) [inline MatrixPilot/MAVLink.c]: as above.

File: Microchip/module.mk [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # this file is included from makefile
 
+ifneq (,$(filter $(TOOLCHAIN), C30 XC16))
+
 extra_dirs := MDD-File-System USB USB/CDC-Device-Driver USB/MSD-Device-Driver
```
Comment (Inspirati) [inline Microchip/module.mk]: I've semi started with moving over to the FatFs library to replace the Microchip library code. Unfortunately as of this moment, it's a lower priority than that which draws my motivation. Also, i'm strongly in favour of keeping the Microchip filesystem library for the AUAV3 bootloader - so perhaps for that particular sub project, we could re-licence it as something akin the the LGPL.

File: Tools/HILSIM_XPlane/Setup.h [hunk 8]
```diff
@@ -161,4 +161,6 @@ class LogFile
 	ofstream mLogFile;
 };
 
-static LogFile LoggingFile;
+//static LogFile LoggingFile;
```
Comment (Inspirati) [inline Tools/HILSIM_XPlane/Setup.h]: Indeed, call this one a lack of clean up at the time.

File: Tools/MatrixPilot-SIL/UDBSocket.h (hunk 1 above)
Comment (Inspirati) [inline Tools/MatrixPilot-SIL/UDBSocket.h]: I left this one here as for the moment I'm without a Windows development machine, and hence have not done any test builds with VC++ for quite some time. When it is tested, should something WRT to this be astray, this 'comment' may add debugging value.

File: libDCM/hilsim.c (hunk 2 above)
Comment (Inspirati) [inline libDCM/hilsim.c]: Have considered this quite extensively, and believe it almost certainly does not belong in libUDB either. Please note the code this module contains was segregated out of libDCM/gpsParseUBX.c, and so if we could consider it an incremental step to improving overall modularity, I'd like to leave it as is for now.

File: libDCM/hilsim.h (hunk 3 above)
Comment (Inspirati) [inline libDCM/hilsim.h]: No doubt something that one needs to be more mindful with many more of our code modules..

**Comment by Inspirati** [comment]
Unless any of the feedback provided above is considered a show stopper, I'd be grateful if we could proceed with incorporating this PR, being mindful of the valuable review insight provided in PR's to come.

**Comment by phollands** [comment]
Hi Robert,

Thanks for all your comments and considerations of my comments. It's good to know where you stand on these things. Merging.

Best wishes, Pete

**Comment by Inspirati** [comment]
Thanks Peter. Whist I am aware that I do tend to get way ahead of even
myself with these kinds of changes, I'm also acutely aware of the fact that
if I don't make a push for integrating such work, even I tend to loose
track of my proposed changes, and in the past so much of it has as such
been lost in the fog. With my limited ability to do proper (read, actual
flight) testing, per review by others is appreciated as being most
important - for which I am truly grateful.

I often feel at a cross-roads between posting massive monolithic changes
(aka, most certainly badly breaking things), and segregating out small
'diffs' which will otherwise seem silly without the greater picture.

Please note much of what I PR is based on being able to port the codebase
to other architectures, and without having many others on-board to help
explain the 'why' of such changes, I fear much of the team may wonder as to
the worth.

**Comment by phollands** [comment]
Robert, I am grateful for all that you do on Code Portability. That is strategic for the future.

With regard to the big picture and vision of the future, that is possibly where we have differed in the past, and where we need better clarity going forward. We need a better consensus on the future of the project.

For now, my main purpose, while I have more time at the moment, is to ensure that the code base is trustworthy for flight.  Hence the large amount of flight testing in the late summer, and resulting changes.
it is vital that all developers in the group know that master is trustworthy. They want to branch off of a trusted base, to make their own feature improvements.

I have been delighted with some of the tools that you have made for checking compilations in Travis, and that is has been a big step forward in ensuring that the code compiles well, with a number of differnt options enabled.

With regard to large code changes, I would hope that we could do that in a branch. Branches can be re-based regularly, and so it should be able to maintain them, and a related PR for quite a long time over many months, until you are ready for the final merge.

One of the nice things about Git and github, is that others (myself) can follow your repositories and branches in detail, and can compile your ideas, and even flight test them before merging to the main repository at MatrixPilot/MatrixPilot. (I am looking forward to Silcat working again)

Best wishes, Pete
